typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -203,36 +208,105 @@ export class EvmERC20WarpRouteReader extends EvmRouterReader {
       destinationGas,
     };
   }
-
-  async deriveWarpRouteVirtualConfig(
-    chain: ChainName,
-    address: Address,
-  ): Promise<HypTokenRouterVirtualConfig> {
-    const virtualConfig: HypTokenRouterVirtualConfig = {
-      contractVerificationStatus: {},
-    };
+  async getContractVerificationStatus(chain: ChainName, address: Address) {
+    const contractVerificationStatus: Record<
+      string,
+      ContractVerificationStatus
+    > = {};
 
     const contractType = (await isProxy(this.provider, address))
       ? VerifyContractTypes.Proxy
       : VerifyContractTypes.Implementation;
 
-    virtualConfig.contractVerificationStatus[contractType] =
+    if (this.multiProvider.isLocalRpc(chain)) {
+      this.logger.debug('Skipping verification for local endpoints');
+      return { [contractType]: ContractVerificationStatus.Skipped };
+    }
+    contractVerificationStatus[contractType] =
       await this.contractVerifier.getContractVerificationStatus(chain, address);
 
     if (contractType === VerifyContractTypes.Proxy) {
-      virtualConfig.contractVerificationStatus.Implementation =
+      contractVerificationStatus[VerifyContractTypes.Implementation] =
         await this.contractVerifier.getContractVerificationStatus(
           chain,
           await proxyImplementation(this.provider, address),
         );
 
       // Derive ProxyAdmin status
-      virtualConfig.contractVerificationStatus.ProxyAdmin =
+      contractVerificationStatus[VerifyContractTypes.ProxyAdmin] =
         await this.contractVerifier.getContractVerificationStatus(
           chain,
           await proxyAdmin(this.provider, address),
         );
     }
+    return contractVerificationStatus;
+  }
+
+  async getOwnerStatus(chain: ChainName, address: Address) {
+    let ownerStatus: Record<string, OwnerStatus> = {};
+    if (this.multiProvider.isLocalRpc(chain)) {
+      this.logger.debug('Skipping owner verification for local endpoints');
+      return {
+        [address]: OwnerStatus.Skipped,
+      };
+    }
+
+    const provider = this.multiProvider.getProvider(chain);
+    const owner = await Ownable__factory.connect(address, provider).owner();
+
+    ownerStatus[owner] = (await isAddressActive(provider, owner))
+      ? OwnerStatus.Active
+      : OwnerStatus.Inactive;
+
+    // Heuristically check if the owner could be a safe by calling an expected function
+    // This status will overwrite 'active' status
+    try {
+      await IOwnerManager__factory.connect(owner, provider).getThreshold();
+      ownerStatus[owner] = OwnerStatus.GnosisSafe;
+    } catch {
+      this.logger.debug(`${owner} is not a safe`);
+    }
```

Comment:
I would check if the nonce method exists too, just to be sure we are not being bamboozled by a contract that coincidentally has the getThreshold method

Reply:
added